sentinel/src/lib/metadata.js

```diff
@@ -1,91 +1,43 @@
 
-const migrateOldMetaDoc = doc => {
-  const stub = {
-    _id: doc._id,
-    _rev: doc._rev,
-    _deleted: true,
-  };
-  logger.info('Deleting old metadata document: %o', doc);
-  return db.medic
-    .put(stub)
-    .then(() => {
-      doc._id = METADATA_DOCUMENT;
-      delete doc._rev;
-      return doc;
-    })
-    .catch(err => {
-      throw err;
-    });
-};
-
-const getMetaData = () => {
-  return db.sentinel.get(METADATA_DOCUMENT).catch(err => {
+const getValue = (docId, defaultValue) => db.sentinel.get(docId)
+  .then(doc => doc.value)
+  .catch(err => {
     if (err.status !== 404) {
       throw err;
     }
-    return db.medic
-      .get(METADATA_DOCUMENT)
-      .then(doc => {
-        // Old doc exists, delete it and return the base doc to be saved later
-        return migrateOldMetaDoc(doc);
-      })
-      .catch(err => {
-        if (err.status !== 404) {
-          throw err;
-        }
-        // Doc doesn't exist.
-        // Maybe we have the doc in the old location?
-        return db.medic
-          .get(OLD_METADATA_DOCUMENT)
-          .then(doc => {
-            // Old doc exists, delete it and return the base doc to be saved later
-            return migrateOldMetaDoc(doc);
-          })
-          .catch(err => {
-            if (err.status !== 404) {
-              throw err;
-            }
-            // No doc at all, create and return default
-            return {
-              _id: METADATA_DOCUMENT,
-              processed_seq: 0,
-            };
-          });
-      });
-  });
-};
 
-const getProcessedSeq = () => {
-  return getMetaData()
-    .then(doc => doc.processed_seq)
-    .catch(err => {
-      logger.error('Error getting meta data: %o', err);
-      throw err;
-    });
-};
+    return defaultValue;
+  });
 
-const updateMetaData = seq => {
-  return getMetaData()
+const setValue = (docId, value) =>
+  db.sentinel.get(docId)
     .then(doc => {
-      doc.processed_seq = seq;
-      return db.sentinel.put(doc).catch(err => {
-        if (err) {
-          logger.error('Error updating metaData: %o', err);
-        }
-      });
+      doc.value = value;
+      return doc;
     })
     .catch(err => {
-      logger.error('Error fetching metaData for update: %o', err);
-      return null;
-    });
-};
+      if (err.status === 404) {
+        return {
+          _id: docId,
+          value: value
+        };
+      }
+
+      throw err;
+    })
+    .then(doc => db.sentinel.put(doc));
+
 
 module.exports = {
-  getProcessedSeq: () => getProcessedSeq(),
-  update: seq => updateMetaData(seq),
+  getTransitionSeq: () => getValue(TRANSITION_SEQ_DOCUMENT, '0'),
+  setTransitionSeq: seq => setValue(TRANSITION_SEQ_DOCUMENT, seq),
+  getBackgroundCleanupSeq: () => getValue(BACKGROUND_CLEANUP_SEQ_DOCUMENT, '0'),
+  setBackgroundCleanupSeq: seq => setValue(BACKGROUND_CLEANUP_SEQ_DOCUMENT, seq),
+  _getValue: getValue,
+  _setValue: setValue
```

Comment:
Consider using rewire here so you don't need to expose these.

Reply:
Actually because these `_` prefixed functions aren't used internally I'd say remove them and test the exposed functions directly.